### PR TITLE
Update libcni to latest version

### PR DESCRIPTION
**Pull request by MikeZappa87**
Updating libcni to support latest version

### CODE REVIEW

**Review by mikebrow — COMMENTED** [review]
Need a TestLibCNIType120 see:
https://github.com/containerd/go-cni/pull/115/files#diff-6cdf85f441267859d068399676e5f0ad8946811bd7e951b4e80db98d9fd2489cL200

may help draw out any issues that may arise as we move up..

Also we should open an issue for the new libcni1.2 changes that need further work..
- GC verb - allows runtimes to specify the set of known-good attachments, allowing plugins to clean up stale and leaked resources such as IPAM reservations. Libcni will also synthesize a CNI DEL for any stale cached attachments, so all users will have a form of GC, even if their plugins do not support CNI v1.1
- The STATUS verb allows a plugin to report its readiness to accept ADD requests. 
- config now includes Version negotiation w/ new cniVersions
  "cniVersion": "1.0.0",
  "cniVersions": ["1.0.0", "1.1.0"]
-GetVersion()
- ** check if version imbalance is an issue now that libcni version 1.20 matches to cni version 1.1.0 .. make sure no issues for the ci/cd/build test/upgrade/downgrade scenarios

File: go.mod [hunk 1]
```diff
@@ -3,11 +3,12 @@ module github.com/containerd/go-cni
 go 1.21
 
 require (
-	github.com/containernetworking/cni v1.1.2
+	github.com/containernetworking/cni v1.2.0
 	github.com/stretchr/testify v1.8.0
 )
 
 require (
+	github.com/Masterminds/semver/v3 v3.2.1 // indirect
```
Comment (mikebrow) [inline go.mod:11]: background discussion with CNI team if exposure of this package was on purpose .. check if a newer tagged version of this semver project is needed etc..

Comment (MikeZappa87) [reply go.mod:11]: @squeed thoughts? We could just revert that pr unless we have a better route

**Comment by MikeZappa87** [comment]
@squeed just want to get this to the top of your email

File: go.mod (hunk 1 above)
Comment (mikebrow) [reply go.mod:11]: Noting nil version equality fix for panic.. https://github.com/Masterminds/semver/pull/213 is not in this tagged version. Would prefer using golang public `[]semver` or `[]string` if possible to reduce externalized package dependencies. 

That said we can handle the two asks separately if desired by the CNI team.

**Review by dims — APPROVED** [review]
LGTM

**Pull request merged**